### ISSUE TITLE
onboard/caribou: crash on accentuated characters

### ISSUE THREAD

**Issue by haggen88** [linuxmint/mint21-beta#28]
obs: Test done in usb live mode. Unfortunately I don't have enough space to check the error on a virtual machine with Mint installed.

Steps to reproduce

1- open a text document
2- open virtual keyboard
3- type a letter that contains other options and select that option  {eg. the n to select the ñ - the c to select the ç}

https://user-images.githubusercontent.com/68134588/179089790-1cde5e68-d779-4f8b-b462-66e91e10990a.mp4

.

[virtual keyboard  log and xsession.zip](https://github.com/linuxmint/mint21-beta/files/9115628/virtual.keyboard.log.and.xsession.zip)

----

P.S.  performing the same steps with Onboard, lead to it being closed

https://user-images.githubusercontent.com/68134588/179090504-ec1e3cb0-ff28-423f-a3e2-9277fb5315a0.mp4

 console output

mint@mint:~$ onboard

(onboard:3138): Gdk-ERROR **: 20:23:23.599: The program 'onboard' received an X Window System error.
This probably reflects a bug in the program.
The error was 'BadValue (integer parameter out of range for operation)'.
  (Details: serial 2958 error_code 2 request_code 135 (XKEYBOARD) minor_code 9)
  (Note to programmers: normally, X errors are reported asynchronously;
   that is, you will receive the error a while after causing it.
   To debug your program, run it with the GDK_SYNCHRONIZE environment
   variable to change this behavior. You can then get a meaningful
   backtrace from your debugger if you break on the gdk_x_error() function.)
Trace/breakpoint trap (core dumped)

[onboard logs.zip](https://github.com/linuxmint/mint21-beta/files/9115644/onboard.logs.zip)

**Comment by JosephMcc** [comment]
I can reproduce the crash with the Virtual Keyboard
```
(cinnamon:580329): Gdk-ERROR **: 11:38:03.062: The program 'cinnamon' received an X Window System error.
This probably reflects a bug in the program.
The error was 'BadValue (integer parameter out of range for operation)'.
  (Details: serial 140302 error_code 2 request_code 135 (XKEYBOARD) minor_code 9)
  (Note to programmers: normally, X errors are reported asynchronously;
   that is, you will receive the error a while after causing it.
   To debug your program, run it with the GDK_SYNCHRONIZE environment
   variable to change this behavior. You can then get a meaningful
   backtrace from your debugger if you break on the gdk_x_error() function.)

```

**Comment by clefebvre** [comment]
The fix is in https://gitlab.freedesktop.org/xorg/xserver/-/commit/cd3d21d8c44a35c22b7eca61a58bb5620b116102.

We'll wait for it to get in 22.04.

We can fix caribou right now by removing the patch, but there's no point in fixing caribou alone if we don't fix onboard as well... without a keyboard you need to enter your password not only for the screensaver and DE but also in the DM (and that uses onboard).

**Comment by clefebvre** [comment]
Crash fixed in caribou 0.4.21+mint4+vanessa.
